src/open_inwoner/accounts/views/login.py

```diff
@@ -0,0 +1,369 @@
+
+        redirect_to = self.request.POST.get(
+            self.redirect_field_name, self.request.GET.get(self.redirect_field_name, "")
+        )
+
+        params = {
+            self.redirect_field_name: redirect_to,
+            "user": signer.sign(user.pk),
+        }
+
+        if not user.phonenumber:
+            # Redirect to add mobile phone number view
+            return HttpResponseRedirect(
+                reverse("add_phone_number") + "?" + urlencode(params)
+            )
+
+        # Two factor: Send SMS
+        # period between changes of the OTP value is in seconds
+        token = totp(
+            user.seed, period=getattr(settings, "ACCOUNTS_USER_TOKEN_EXPIRE_TIME", 300)
+        )
+
+        self.request.user = user
+
+        try:
+            gateway.send(to=user.phonenumber, token=token)
+        except GatewayError:
+            messages.error(
+                self.request,
+                "Het is vanwege een storing tijdelijk niet mogelijk om "
+                "in te loggen, probeer het over 15 minuten nogmaals. "
+                "Mocht het inloggen na meerdere pogingen niet werken, "
+                "neem dan contact op met de Open Inwoner Platform.",
+            )
+            self.log_user_action(
+                user,
+                f"Het versturen van een SMS-bericht aan {user.phonenumber} is mislukt. Inloggen afgebroken.",
+            )
+            return HttpResponseRedirect(reverse("pages-root"))
+        else:
+            self.log_user_action(
+                user,
+                "SMS bericht met code is verzonden aan {0}".format(user.phonenumber),
+            )
+
+            messages.debug(self.request, gateway.get_message(token))
+
+        return HttpResponseRedirect(reverse("verify_token") + "?" + urlencode(params))
+
+
+class VerifyTokenView(ThrottleMixin, FormView):
+    throttle_visits = 3
+    throttle_period = 60
+    throttle_403 = False
+    throttle_name = "verify_token"
+
+    template_name = "registration/verify_token.html"
+    form_class = VerifyTokenForm
+    redirect_field_name = REDIRECT_FIELD_NAME
+    sms_button_text = _("Verstuur SMS")
+    message = _(
+        "U ontvangt binnen 1 minuut een sms-bericht op uw mobiele telefoon. "
+        "Vul de code die in het bericht staat hieronder in en klik op "
+        "<em>Inloggen</em>. De code is maximaal 5 minuten geldig. "
+        "Mocht het niet lukken om de code binnen deze tijd in te voeren, klik "
+        "dan op <em>Verstuur nieuwe SMS</em>."
+    )
+    sms_button_text = _("Verstuur nieuwe SMS")
+
+    def log_in(self):
+        """
+        After security check is completed, log the user in.
+        """
+        auth_login(
+            self.request,
+            self.user_cache,
+            backend="open_inwoner.accounts.backends.UserModelEmailBackend",
+        )
+
+    @method_decorator(never_cache)
+    def dispatch(self, request, *args, **kwargs):
+        UserModel = get_user_model()
```

Comment:
In this project we don't support swappable User models so you don't need `get_user_model()` an can just hard-code to the real model.